Library/ViewModels/BackerDashboardProjectCellViewModel.swift

```diff
@@ -0,0 +1,152 @@
+import Foundation
+import KsApi
+import ReactiveSwift
+import Result
+
+public protocol BackerDashboardProjectCellViewModelInputs {
+  /// Call with a backed project.
+  func project(_ project: Project)
+}
+
+public protocol BackerDashboardProjectCellViewModelOutputs {
+  /// Emits project name and state for screen reader.
+  var cellAccessibilityLabel: Signal<String, NoError> { get }
+
+  /// Emits a boolean whether to show the metadata timer icon.
+  var metadataIconIsHidden: Signal<Bool, NoError> { get }
+
+  /// Emits text for the metadata label.
+  var metadataText: Signal<String, NoError> { get }
+
+  /// Emits attributed text for the percent funded label.
+  var percentFundedText: Signal<NSAttributedString, NoError> { get }
+
+  /// Emits the project's photo URL to be displayed.
+  var photoURL: Signal<URL?, NoError> { get }
+
+  /// Emits the project's funding progress amount to be displayed.
+  var progress: Signal<Float, NoError> { get }
+
+  /// Emits a color for the progress bar.
+  var progressBarColor: Signal<UIColor, NoError> { get }
+
+  /// Emits the project name to be displayed.
+  var projectTitleText: Signal<NSAttributedString, NoError> { get }
+
+  /// Emits a boolean when the saved icon is hidden or not.
+  var savedIconIsHidden: Signal<Bool, NoError> { get }
+}
+
+public protocol BackerDashboardProjectCellViewModelType {
+  var inputs: BackerDashboardProjectCellViewModelInputs { get }
+  var outputs: BackerDashboardProjectCellViewModelOutputs { get }
+}
+
+public final class BackerDashboardProjectCellViewModel: BackerDashboardProjectCellViewModelType,
+  BackerDashboardProjectCellViewModelInputs, BackerDashboardProjectCellViewModelOutputs {
+  public init() {
+    let project = projectProperty.signal.skipNil()
+
+    self.projectTitleText = project.map(titleString(for:))
+
+    self.photoURL = project.map { URL(string: $0.photo.full) }
+
+    self.progress = project.map { $0.stats.fundingProgress }
+
+    self.cellAccessibilityLabel = project.map { "\($0.name) \($0.state.rawValue)" }
+
+    self.metadataText = project.map(metadataString(for:))
+
+    self.metadataIconIsHidden = project.map { $0.state != .live }
+
+    self.percentFundedText = project.map(percentFundedString(for:))
+
+    self.progressBarColor = project.map(progressBarColor(for:))
+
+    self.savedIconIsHidden = project.map { $0.personalization.isStarred != .some(true) ?? true }
```

Comment:
no need to coalesce `?? true` since `$0.personalization.isStarred != .some(true)` returns an honest boolean.

woulda thought swift would warn about that tbh